### PR TITLE
Core: Deprecate immutable options as parameters

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: #11142

## What I did

Deprecated

## How to test

Add deprecated parameter to official SB

### CODE REVIEW

File: lib/api/src/modules/stories.ts [hunk 1]
```diff
@@ -26,7 +27,7 @@ import {
   SetStoriesPayloadV2,
 } from '../lib/stories';
 
-import { Args, ModuleFn } from '../index';
+import { Args, ModuleFn, useAddonState } from '../index';
```
Comment (shilman) [inline lib/api/src/modules/stories.ts:30]: Unnecessary import?

**Review by shilman — CHANGES_REQUESTED** [review]
LGTM!

File: lib/api/src/modules/stories.ts (hunk 1 above)
Comment (tmeasday) [reply lib/api/src/modules/stories.ts]: Where did that come from? Must have been VS code being too smart for its own good.

**Review by shilman — APPROVED** [review]
LGTM!

**Review by shilman — CHANGES_REQUESTED** [review]
can you fix the failing tests?

**Comment by tmeasday** [comment]
@shilman passing now I think

**Pull request merged**